contracts/evoting/evoting.go

```diff
@@ -783,36 +797,79 @@ func (e evotingCommand) deleteForm(snap store.Snapshot, step execution.Step) err
 	return nil
 }
 
-// manageAdminForm implements commands. It performs the ADD or REMOVE ADMIN command
-func (e evotingCommand) manageAdminForm(snap store.Snapshot, step execution.Step) error {
+// manageAdminList implements commands. It performs the ADD or REMOVE ADMIN command
+func (e evotingCommand) manageAdminList(snap store.Snapshot, step execution.Step) error {
 	msg, err := e.getTransaction(step.Current)
 	if err != nil {
 		return xerrors.Errorf(errGetTransaction, err)
 	}
 
-	var form types.AdminForm
-	var formID []byte
+	var form types.AdminList
+
+	h := sha256.New()
+	h.Write([]byte(AdminListId))
+	formIDBuf := h.Sum(nil)
 
 	txAddAdmin, okAddAdmin := msg.(types.AddAdmin)
 	txRemoveAdmin, okRemoveAdmin := msg.(types.RemoveAdmin)
 
 	if okAddAdmin {
-		form, formID, err = e.getAdminForm(txAddAdmin.FormID, snap)
+		// Recover the AdminList stored on the blockchain
+		form, err = e.getAdminList(snap)
 		if err != nil {
-			return xerrors.Errorf("failed to get AdminForm: %v", err)
+			if !strings.Contains(err.Error(), "No form found") {
```

Comment:
is it possible to match the exact error string here? if someone else would create an error message like "No form found because of some other error" it would not match despite the list possibly existing

that any error message starting by "No form found" will lead to the code trying to create the chain is implicit knowledge that makes it hard to debug

alternatively you could use a custom error

Reply:
I've pushed a new version with a matching on the exact error string.

Then as it is with a Not in front, basically as soon as it contains "No form found" it blocks in a fail safe by avoiding it.